lib/core.coffee

```
@@ -1,102 +1,66 @@
+  # if window is defined, then it must be running in a browser. Atom breaks
+  # this assumption, so CoffeeLint < 1.9.1 will fail to find CoffeeScript.
+  # See https://github.com/clutchski/coffeelint/pull/383
+  if semver.lt(coffeelint.VERSION, '1.9.1')
+    coffeeLintPath = 'coffeelint'
     coffeelint = require(coffeeLintPath)
+    showUpgradeError = true
 
-    # Versions before 1.9.1 don't work with atom because of an assumption that
-    # if window is defined, then it must be running in a browser. Atom breaks
-    # this assumption, so CoffeeLint < 1.9.1 will fail to find CoffeeScript.
-    # See https://github.com/clutchski/coffeelint/pull/383
-    [major, minor, patch] = coffeelint.VERSION.split('.').map(toInt)
-    if not @isCompatibleWithAtom(coffeelint)
-      coffeeLintPath = 'coffeelint'
-      coffeelint = require(coffeeLintPath)
-      showUpgradeError = true
-
-    configFinder = require("#{coffeeLintPath}/lib/configfinder")
+  configFinder = require("#{coffeeLintPath}/lib/configfinder")
 
-    result = []
-    try
-      config = configFinder.getConfig(filePath)
-      if @configImportsModules(config) and not @canImportModules(coffeelint)
-        showUpgradeError = true
-      else
-        result = coffeelint.lint(source, config, isLiterate)
-    catch e
-      console.log(e.message)
-      console.log(e.stack)
-      result.push({
-        lineNumber: 1
-        level: 'error'
-        message: "CoffeeLint crashed, see console for error details."
-        rule: 'none'
-      })
+  result = []
+  try
+    config = configFinder.getConfig(filePath)
+    if configImportsModules(config) and semver.lt(coffeelint.VERSION, '1.9.5')
+      showUpgradeError = true
+    else
+      result = coffeelint.lint(source, config, isLiterate)
+  catch e
+    console.log(e.message)
+    console.log(e.stack)
+    result.push({
+      lineNumber: 1
+      level: 'error'
+      message: 'CoffeeLint crashed, see console for error details.'
+      rule: 'none'
```

I'm wondering if we should just let this throw a error instead of showing this on line 1